BUG: SIGSEV when running terraform plan if endpoint is not reachable

Hi, I ran into issues when running terraform plan.

If the API is not reachable the provider crashes instead of printing some useful information.

```sh
terraform plan

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Request cancelled
│
│ The plugin6.(*GRPCProvider).ConfigureProvider request was cancelled.
╵

Stack trace from the terraform-provider-sonatyperepo_v0.6.2 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x48 pc=0x10357744c]

goroutine 33 [running]:
terraform-provider-sonatyperepo/internal/provider/common.getResponseBody(0x0)
	terraform-provider-sonatyperepo/internal/provider/common/api.go:42 +0x1c
terraform-provider-sonatyperepo/internal/provider/common.HandleApiError({0x103910734, 0x38}, 0x140003651c8, 0x0, 0x1400039e420)
	terraform-provider-sonatyperepo/internal/provider/common/api.go:30 +0x38
terraform-provider-sonatyperepo/internal/provider/common.(*SonatypeDataSourceData).CheckWritableAndGetVersion(0x140003652f8, {0x103cd7d08, 0x1400039cea0}, 0x1400039e420)
	terraform-provider-sonatyperepo/internal/provider/common/model.go:43 +0x2ec
terraform-provider-sonatyperepo/internal/provider.(*SonatypeRepoProvider).Configure(0x140001d5930, {0x103cd7d08, 0x1400039cea0}, {{0x140003940b0, 0x6}, {{{0x103cdcda8, 0x1400039d710}, {0x103b5a4e0, 0x1400039d5c0}}, {0x103cdffc8, ...}}, ...}, ...)
	terraform-provider-sonatyperepo/internal/provider/provider.go:182 +0x85c
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ConfigureProvider(0x140002e2008, {0x103cd7d08, 0x1400039cea0}, 0x1400039e460, 0x1400039e410)
	github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_configureprovider.go:18 +0xe0
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ConfigureProvider(0x140002e2008, {0x103cd7d08?, 0x1400039cdb0?}, 0x140003a61e0)
	github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/proto6server/server_configureprovider.go:39 +0x288
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ConfigureProvider(0x14000213040, {0x103cd7d08?, 0x1400039c1e0?}, 0x1400039e0f0)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/tf6server/server.go:559 +0x2ac
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ConfigureProvider_Handler({0x103c97640, 0x14000213040}, {0x103cd7d08, 0x1400039c1e0}, 0x140003b0000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:557 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400020a200, {0x103cd7d08, 0x1400039c150}, 0x140003880c0, 0x140002da8d0, 0x10445cb10, 0x0)
	google.golang.org/grpc@v1.73.0/server.go:1405 +0xca8
google.golang.org/grpc.(*Server).handleStream(0x1400020a200, {0x103cd85d8, 0x1400037e000}, 0x140003880c0)
	google.golang.org/grpc@v1.73.0/server.go:1815 +0x910
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.73.0/server.go:1035 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 23
	google.golang.org/grpc@v1.73.0/server.go:1046 +0x13c

Error: The terraform-provider-sonatyperepo_v0.6.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/internal/provider/common/api.go
+++ b/internal/provider/common/api.go
@@ -16,19 +16,31 @@
 package common
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"io"
 	"log"
+	"net"
 	"net/http"
+	"syscall"
 
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 )
 
 func HandleApiError(message string, err *error, httpResponse *http.Response, respDiags *diag.Diagnostics) {
-	respDiags.AddError(
-		message,
-		fmt.Sprintf("%s: %s: %s", *err, httpResponse.Status, getResponseBody(httpResponse)),
-	)
+	networkError, errorMessage := handleNetworkError(*err)
+	if networkError {
+		respDiags.AddError(
+			errorMessage,
+			fmt.Sprintf("Networking Error: %s (%v)", errorMessage, *err),
+		)
+	} else {
+		respDiags.AddError(
+			message,
+			fmt.Sprintf("%s: %s: %s", *err, httpResponse.Status, getResponseBody(httpResponse)),
+		)
+	}
 }
 
 func HandleApiWarning(message string, err *error, httpResponse *http.Response, respDiags *diag.Diagnostics) {
@@ -36,6 +48,31 @@ func HandleApiWarning(message string, err *error, httpResponse *http.Response, r
 		message,
 		fmt.Sprintf("%s: %s: %s", *err, httpResponse.Status, getResponseBody(httpResponse)),
 	)
+}
+
+func handleNetworkError(err error) (bool, string) {
+	// Check for specific error types
+	if opErr, ok := err.(*net.OpError); ok {
+		// Network operation error (dial, read, write, etc.)
+		return true, fmt.Sprintf("OpError: %s, %s", opErr.Op, opErr.Net)
+	}
+
+	if dnsErr, ok := err.(*net.DNSError); ok {
+		// DNS resolution error
+		return true, fmt.Sprintf("DNS Error: %v", dnsErr)
+	}
+
+	if errors.Is(err, context.DeadlineExceeded) {
+		// Timeout error
+		return true, fmt.Sprintf("Connection timed out: %v", err)
+	}
+
+	// General network error check
+	if errors.Is(err, syscall.ECONNREFUSED) {
+		return true, fmt.Sprintf("Connection refused: %v", err)
+	}
+
+	return false, ""
 }
 
 func getResponseBody(httpResponse *http.Response) []byte {

